skia/recorder/src/main.cpp

```diff
@@ -0,0 +1,148 @@
+#include "SkData.h"
+#include "SkImage.h"
+#include "SkStream.h"
+#include "SkSurface.h"
+#include "animation/animation.hpp"
+#include "animation/linear_animation.hpp"
+#include "args.hxx"
+#include "artboard.hpp"
+#include "core/binary_reader.hpp"
+#include "file.hpp"
+#include "math/aabb.hpp"
+#include "skia_renderer.hpp"
+#include <cstdio>
+#include <iostream>
+#include <memory>
+#include <stdexcept>
+#include <stdio.h>
+#include <string>
+
+extern "C"
+{
+#include <libavcodec/avcodec.h>
+#include <libavcodec/avfft.h>
+
+#include <libavfilter/buffersink.h>
+#include <libavfilter/buffersrc.h>
+
+#include <libavformat/avformat.h>
+#include <libavformat/avio.h>
+
+#include <libavutil/channel_layout.h>
+#include <libavutil/common.h>
+#include <libavutil/file.h>
+#include <libavutil/imgutils.h>
+#include <libavutil/mathematics.h>
+#include <libavutil/opt.h>
+#include <libavutil/pixdesc.h>
+#include <libavutil/samplefmt.h>
+#include <libavutil/time.h>
+
+#include <libswscale/swscale.h>
+}
+
+#include "args.hxx"
+#include "extractor.hpp"
+#include "writer.hpp"
+#include <fstream>
+#include <string>
+
+#ifdef TESTING
+#else
+int main(int argc, char* argv[])
+{
+	// PARSE INPUT
+	args::ArgumentParser parser(
+	    "Record playback of a Rive file as a movie, gif, etc (eventually "
+	    "should support image sequences saved in a zip/archive too).",
+	    "Experimental....");
+	args::HelpFlag help(
+	    parser, "help", "Display this help menu", {'h', "help"});
+	args::Group required(
+	    parser, "required arguments:", args::Group::Validators::All);
+	args::Group optional(
+	    parser, "optional arguments:", args::Group::Validators::DontCare);
+
+	args::ValueFlag<std::string> source(
+	    required, "path", "source filename", {'s', "source"});
+	args::ValueFlag<std::string> destination(
+	    required, "path", "destination filename", {'d', "destination"});
+	args::ValueFlag<std::string> animationOption(
+	    optional,
+	    "name",
+	    "animation to be played, determines the numbers of frames recorded",
+	    {'a', "animation"});
+	args::ValueFlag<std::string> artboardOption(
+	    optional, "name", "artboard to draw from", {'t', "artboard"});
+	args::ValueFlag<std::string> watermarkOption(
+	    optional, "path", "watermark filename", {'w', "watermark"});
+	args::CompletionFlag completion(parser, {"complete"});
+	try
+	{
+		parser.ParseCLI(argc, argv);
+	}
+	catch (const args::Completion& e)
+	{
+		std::cout << e.what();
+		return 0;
+	}
+	catch (const args::Help&)
+	{
+		std::cout << parser;
+		return 0;
+	}
+	catch (const args::ParseError& e)
+	{
+		std::cerr << e.what() << std::endl;
+		std::cerr << parser;
+		return 1;
+	}
+	catch (args::ValidationError e)
+	{
+		std::cerr << e.what() << std::endl;
+		std::cerr << parser;
+		return 1;
+	}
+
+	RiveFrameExtractor* extractor;
+	try
+	{
+		extractor = new RiveFrameExtractor(args::get(source).c_str(),
```

Comment:
When deleting this, make sure that also early-outs are handled (in `catch` blocks for example)